editor/modes/visual-editor/block.js

```diff
@@ -147,8 +147,22 @@ class VisualEditorBlock extends Component {
 		}
 	}
 
-	stopTyping() {
+	stopTypingOnMouseMove( event ) {
+		// We need to check that the mouse really moved
+		// Because Safari trigger mousemove event when we press shift, ctrl...
+		if ( ! this.lastClientX || ! this.lastClientY ) {
+			this.lastClientX = event.clientX;
+			this.lastClientY = event.clientY;
+			return;
+		}
+
+		if ( this.lastClientX === event.clientX && this.lastClientY === event.clientY ) {
+			return;
+		}
+
 		this.props.onStopTyping();
+		this.lastClientX = event.clientX;
+		this.lastClientY = event.clientY;
```

Comment:
How about this to avoid duplication assignment?

```js
stopTypingOnMouseMove( { clientX, clientY } ) {
	const { lastClientX, lastClientY } = this;

	// We need to check that the mouse really moved
	// Because Safari trigger mousemove event when we press shift, ctrl...
	if ( lastClientX && lastClientY && ( lastClientX !== clientX || lastClientY !== clientY ) ) {
		this.props.onStopTyping();
	}

	this.lastClientX = clientX;
	this.lastClientY = clientY;
}
```